sqlite/db.go

```diff
@@ -0,0 +1,46 @@
+package sqlite
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/mutecomm/go-sqlcipher" // We require go sqlcipher that overrides default implementation
+)
+
+// The reduced number of kdf iterations (for performance reasons) which is
+// currently used for derivation of the database key
+// https://github.com/status-im/status-go/pull/1343
+// https://notes.status.im/i8Y_l7ccTiOYq09HVgoFwA
+const kdfIterationsNumber = 3200
+
+func openDB(path string, key string, kdfIter int) (*sql.DB, error) {
+	db, err := sql.Open("sqlite3", path)
+	if err != nil {
+		return nil, err
+	}
+
+	keyString := fmt.Sprintf("PRAGMA key = '%s'", key)
+
+	// Disable concurrent access as not supported by the driver
+	db.SetMaxOpenConns(1)
+
+	if _, err = db.Exec("PRAGMA foreign_keys=ON"); err != nil {
+		return nil, err
+	}
+
+	if _, err = db.Exec(keyString); err != nil {
+		return nil, err
+	}
+
+	kdfString := fmt.Sprintf("PRAGMA kdf_iter = '%d'", kdfIter)
```

Comment:
like in the previous one, is there any reason why we use `fmt.Sprintf` here instead of doing like here: https://github.com/status-im/status-go/pull/1443/files#diff-dad4e25e26c8c6c84ee658b3920ce82cR80

Reply:
okay, this one isn't very critical, but a comment about not being able to use ? in PRAGMA statements would be nice here and in `keyString`